### PR TITLE
fix/labels-with-url

### LINKED ISSUE #2744: [UI] Label containing a URL cannot be used to filter Executions
### Explain the bug

Screenshare showing the issue https://share.descript.com/view/cBT4aNM66ur 

Reproducer:

```yaml
id: labels
namespace: dev

labels:
  user: anna-geller
  url: "https://github.com/kestra-io/kestra/issues/2632"

tasks:
  - id: log
    type: io.kestra.core.tasks.log.Log
    message: hello there
```

image

Server stack trace:
```
2023-12-19 13:47:45,140 ERROR io-executor-thread-1 i.k.w.controllers.ErrorController Invalid queryString parameter
io.micronaut.http.exceptions.HttpStatusException: Invalid queryString parameter
	at io.kestra.webserver.utils.RequestUtils.lambda$toMap$0(RequestUtils.java:18)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at io.kestra.webserver.utils.RequestUtils.toMap(RequestUtils.java:26)
	at io.kestra.webserver.controllers.ExecutionController.find(ExecutionController.java:160)
	at io.kestra.webserver.controllers.$ExecutionController$Definition$Intercepted.$$access$$find(Unknown Source)
	at io.kestra.webserver.controllers.$ExecutionController$Definition$Exec.dispatch(Unknown Source)
	at io.micronaut.context.AbstractExecutableMethodsDefinition$DispatchedExecutableMethod.invoke(AbstractExecutableMethodsDefinition.java:371)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:128)
	at io.micronaut.validation.ValidatingInterceptor.intercept(ValidatingInterceptor.java:143)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:137)
	at io.kestra.webserver.controllers.$ExecutionController$Definition$Intercepted.find(Unknown Source)
	at io.kestra.webserver.controllers.$ExecutionController$Definition$Exec.dispatch(Unknown Source)
	at io.micronaut.context.AbstractExecutableMethodsDefinition$DispatchedExecutableMethod.invoke(AbstractExecutableMethodsDefinition.java:371)
	at io.micronaut.context.DefaultBeanContext$4.invoke(DefaultBeanContext.java:594)
	at io.micronaut.web.router.AbstractRouteMatch.execute(AbstractRouteMatch.java:303)
	at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:111)
	at io.micronaut.http.context.ServerRequestContext.with(ServerRequestContext.java:103)
	at io.micronaut.http.server.RouteExecutor.lambda$executeRoute$14(RouteExecutor.java:659)
	at reactor.core.publisher.FluxDeferContextual.subscribe(FluxDeferContextual.java:49)
	at reactor.core.publisher.InternalFluxOperator.subscribe(InternalFluxOperator.java:62)
	at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.run(FluxSubscribeOn.java:194)
	at io.micronaut.reactive.reactor.instrument.ReactorInstrumentation.lambda$init$0(ReactorInstrumentation.java:62)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
	at io.micrometer.core.instrument.composite.CompositeTimer.recordCallable(CompositeTimer.java:129)
	at io.micrometer.core.instrument.Timer.lambda$wrap$1(Timer.java:206)
	at io.micronaut.scheduling.instrument.InvocationInstrumenterWrappedCallable.call(InvocationInstrumenterWrappedCallable.java:53)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
```

### Environment Information

- Kestra Version: develop
- Operating System and Java Version (if not using Kestra Docker image):

### PR DESCRIPTION
**Pull request by brian-mulier-p**
closes #2744

### CODE REVIEW

File: webserver/src/main/java/io/kestra/webserver/utils/RequestUtils.java [hunk 1]
```diff
@@ -13,14 +14,14 @@ public static Map<String, String> toMap(List<String> queryString) {
...
-                if (split.length != 2) {
+                String[] split = s.split("[: ]+");
+                if (split.length < 2) {
                     throw new HttpStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "Invalid queryString parameter");
                 }
 
                 return new AbstractMap.SimpleEntry<>(
                     split[0],
-                    split[1]
+                    String.join(":", Arrays.copyOfRange(split, 1, split.length))
```
Comment (loicmathieu) [inline webserver/src/main/java/io/kestra/webserver/utils/RequestUtils.java:24]: This is the same as using indexOf(':') and taking what' s before as key and what's after as value right?

Comment (loicmathieu) [reply webserver/src/main/java/io/kestra/webserver/utils/RequestUtils.java:24]: By doing that we miss some protection that we have before with miss-crafted query string

File: webserver/src/test/java/io/kestra/webserver/controllers/ExecutionControllerTest.java [hunk 2]
```diff
@@ -52,6 +53,8 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 class ExecutionControllerTest extends JdbcH2ControllerTest {
+    public static final String URL_LABEL_VALUE = "https://some-url.com";
+    public static final String ENCODED_URL_LABEL_VALUE = URL_LABEL_VALUE.replace("/", URLEncoder.encode("/", StandardCharsets.UTF_8));
```
Comment (loicmathieu) [inline webserver/src/test/java/io/kestra/webserver/controllers/ExecutionControllerTest.java:57]:
```suggestion
    public static final String ENCODED_URL_LABEL_VALUE = URLEncoder.encode(URL_LABEL_VALUE, StandardCharsets.UTF_8));
```

The all string must be encoded

**Review by loicmathieu — CHANGES_REQUESTED** [review]

File: webserver/src/test/java/io/kestra/webserver/controllers/ExecutionControllerTest.java (hunk 2 above)
Comment (brian-mulier-p) [reply webserver/src/test/java/io/kestra/webserver/controllers/ExecutionControllerTest.java:57]: That's what I did before, I forgot a comment but I saw that this is the behaviour of our UI library (for eg. ":" aren't encoded from UI) so I prefer to replicate this encoding to ensure we don't miss any side-effect (":" parsing in this example)

File: webserver/src/main/java/io/kestra/webserver/utils/RequestUtils.java (hunk 1 above)
Comment (brian-mulier-p) [reply webserver/src/main/java/io/kestra/webserver/utils/RequestUtils.java:24]: I don't think we should handle the syntax at this level. The goal of this method should only be to extract before & after first ":" imo.
For the indexOf yes I can also do that :+1:

Comment (loicmathieu) [reply webserver/src/main/java/io/kestra/webserver/utils/RequestUtils.java:24]: So in this case indexOf is straightforward

File: webserver/src/test/java/io/kestra/webserver/controllers/ExecutionControllerTest.java (hunk 2 above)
Comment (loicmathieu) [reply webserver/src/test/java/io/kestra/webserver/controllers/ExecutionControllerTest.java:57]: We should URL encode everything from the UI, this is a bug and a security risk I think

**Review by loicmathieu — APPROVED** [review]
LGTM

**Pull request merged**